pros/cli/common.py

```diff
@@ -194,28 +194,41 @@ def pros_root(f):
     return decorator
 
 
-def resolve_v5_port(port: Optional[str], type: str, quiet: bool = False) -> Optional[str]:
+def resolve_v5_port(port: Optional[str], type: str, quiet: bool = False) -> Tuple[Optional[str], bool]:
+    """
+    Detect serial ports that can be used to interact with a V5.
+
+    Returns a tuple of (port?, is_joystick). port will be None if no ports are
+    found, and is_joystick is False unless type == 'user' and the port is
+    determined to be a controller. This is useful in e.g.
+    pros.cli.terminal:terminal where the communication protocol is different for
+    wireless interaction.
+    """
     from pros.serial.devices.vex import find_v5_ports
+    # If a port is specified manually, we'll just assume it's
+    # not a joystick.
+    is_joystick = False
     if not port:
         ports = find_v5_ports(type)
         if len(ports) == 0:
             if not quiet:
                 logger(__name__).error('No {0} ports were found! If you think you have a {0} plugged in, '
                                        'run this command again with the --debug flag'.format('v5'),
                                        extra={'sentry': False})
-            return None
+            return None, False
         if len(ports) > 1:
             if not quiet:
                 port = click.prompt('Multiple {} ports were found. Please choose one: '.format('v5'),
                                     default=ports[0].device,
                                     type=click.Choice([p.device for p in ports]))
                 assert port in [p.device for p in ports]
             else:
-                return None
+                return None, False
         else:
             port = ports[0].device
+            is_joystick = type == 'user' and 'Controller' in ports[0].description
```

Comment:
Who controls description? As far as I can tell, it's something the device/os does. I'm worried that this will break across devices/different oses.

Reply:
this is how we've always done it. see here (in particular `filter_v5_ports`): https://github.com/purduesigbots/pros-cli/blob/4460a4a14487efd2f773ad38ad324cab10a5d16d/pros/serial/devices/vex/v5_device.py#L32-L81 for a taste of the blood, sweat, and tears that went into port detection in the first place.

since kunwar tested this (presumably on his macbook), i'm more willing to say it probably works in the majority of cases, given that in the past macOS has been the most troublesome. if @WillXuCodes can test at some point too for extra verification, that might help assuage fears of incompatibility